prog/dftb+/lib_reks/reksgrad.F90

```diff
@@ -2311,9 +2310,7 @@ subroutine RTshift(env, sccCalc, denseDesc, neighbourList, nNeighbourSK, &
     end if
 
     deriv1(:,:,:) = 0.0_dp
-    if (tNAC) then
-      deriv2(:,:,:) = 0.0_dp
-    end if
+    deriv2(:,:,:) = 0.0_dp
```

Comment:
Here, the size of `deriv2` is not too large, thus it do not increase memory allocation so much although we always allocate the `deriv2`. I think that it do not cause the problems when we always allocate the `deriv2`, but should we remove the if condition for this line?

Reply:
This was causing a problem for cases where tNAC was false but the various later changes of (out) to (inout) removed the problem, so the if could be returned. 
However it is now more consistent with the the other variables which initialise to zero at higher levels and, as you mention, is always allocated.